snapshot/pkg/cloudprovider/providers/openstack/openstack_routes.go

```diff
@@ -125,7 +125,7 @@ func updateAllowedAddressPairs(network *gophercloud.ServiceClient, port *neutron
 	origPairs := port.AllowedAddressPairs // shallow copy
 
 	_, err := neutronports.Update(network, port.ID, neutronports.UpdateOpts{
-		AllowedAddressPairs: &newPairs,
+		AllowedAddressPairs: newPairs,
```

Comment:
Same here.  According to the following code, this should be a pointer so the original code is correct.  Is the definition in the SDK different?
https://github.com/gophercloud/gophercloud/blob/master/openstack/networking/v2/ports/requests.go#L118
		AllowedAddressPairs *[]AddressPair `json:"allowed_address_pairs,omitempty"`